From: =?utf-8?b?T25kxZllaiBQb2hvxZllbHNrw70=?= <opohorel@redhat.com>
Date: Tue, 28 Nov 2023 14:33:43 +0100
Subject: wifi: Add `6GHz` to `Supported Frequencies`

Adds `6GHz` to `Supported Frequencies` in Wi-Fi Details page.

Also removes `g_strdup()`, since the string is only passed to
`gtk_label_set_label()` anyways.

Resolves: #2474
---
 panels/network/connection-editor/ce-page-details.c | 19 +++++++++++++++----
 1 file changed, 15 insertions(+), 4 deletions(-)

diff --git a/panels/network/connection-editor/ce-page-details.c b/panels/network/connection-editor/ce-page-details.c
index 534c01a..7d863a9 100644
--- a/panels/network/connection-editor/ce-page-details.c
+++ b/panels/network/connection-editor/ce-page-details.c
@@ -281,7 +281,7 @@ connect_details_page (CEPageDetails *self)
         const gchar *type;
         const gchar *hw_address = NULL;
         g_autofree gchar *security_string = NULL;
-        g_autofree gchar *freq_string = NULL;
+        const gchar *freq_string = NULL;
         const gchar *strength_label;
         gboolean device_is_active;
         NMIPConfig *ipv4_config = NULL, *ipv6_config = NULL;
@@ -342,13 +342,24 @@ connect_details_page (CEPageDetails *self)
         gtk_widget_set_visible (GTK_WIDGET (self->mac_label), hw_address != NULL);
 
         if (wifi_caps & NM_WIFI_DEVICE_CAP_FREQ_VALID) {
+/* Check 6 GHz support in Network Manager */
+#if NM_CHECK_VERSION (1, 45, 4)
+                if (wifi_caps & NM_WIFI_DEVICE_CAP_FREQ_2GHZ &&
+                    wifi_caps & NM_WIFI_DEVICE_CAP_FREQ_5GHZ &&
+                    wifi_caps & NM_WIFI_DEVICE_CAP_FREQ_6GHZ)
+                        freq_string = _("2.4 GHz / 5 GHz / 6 GHz");
+                else if (wifi_caps & NM_WIFI_DEVICE_CAP_FREQ_5GHZ &&
+                         wifi_caps & NM_WIFI_DEVICE_CAP_FREQ_6GHZ)
+                        freq_string = _("5 GHz / 6 GHz");
+                else
+#endif
                 if (wifi_caps & NM_WIFI_DEVICE_CAP_FREQ_2GHZ &&
                     wifi_caps & NM_WIFI_DEVICE_CAP_FREQ_5GHZ)
-                        freq_string = g_strdup (_("2.4 GHz / 5 GHz"));
+                        freq_string = _("2.4 GHz / 5 GHz");
                 else if (wifi_caps & NM_WIFI_DEVICE_CAP_FREQ_2GHZ)
-                        freq_string = g_strdup (_("2.4 GHz"));
+                        freq_string = _("2.4 GHz");
                 else if (wifi_caps & NM_WIFI_DEVICE_CAP_FREQ_5GHZ)
-                        freq_string = g_strdup (_("5 GHz"));
+                        freq_string = _("5 GHz");
         }
 
         gtk_label_set_label (self->freq_label, freq_string);
